### PR TITLE
Fixed float based placing bug & comparatorBetterItemFrames

### PR DESCRIPTION
**Pull request by fxmorin**
When placing pistons, observers, etc... past 2^24 every 3 blocks they will face down. This is because float size issue, made it a double to fix it.

### CODE REVIEW

**Comment by fxmorin** [comment]
Untested

**Comment by fxmorin** [comment]
I have tested the double-based placing bug. My fix works correctly.
I have also added the `comparatorBetterItemFrames` rule

**Comment by altrisi** [comment]
Please follow naming conventions for the enum class! (`ComparatorOptions`). Also, if adding to an existing mixin, either rename it so it's not `comparatorReadsClock` or add a new one.

About the comparators, is the replacing of basically everything in the method necessary? Seems like some is a copy of vanilla, mostly because of the names of the variables that seem to come from a decompiler (this is mostly a thing of mine, may not be required for merging).

**Comment by fxmorin** [comment]
I am not overriding or using a Redirect due to the fact I also do not like restricting mod compatibility. Although I did have to Inject and basically rewrite the comparator due to not having good enough injection points.

File: src/main/resources/carpet-extra.mixins.json [hunk 1]
```diff
@@ -3,73 +3,71 @@
   "package": "carpetextra.mixins",
   "compatibilityLevel": "JAVA_8",
   "mixins": [
-    "DispenserBlockMixin",
```
Comment (gnembon) [inline src/main/resources/carpet-extra.mixins.json:6]: Please don't sort mixins unnecessarily

Comment (fxmorin) [reply src/main/resources/carpet-extra.mixins.json:6]: Oops didn't notice the plugin was doing that. I'll fix it in a bit

**Review by gnembon — APPROVED** [review]

**Pull request merged**